projects/jsf/src/lib/models/integer-data-item.ts

```diff
@@ -0,0 +1,49 @@
+import { SchemaIntegerOptions } from '../form-data-item.service';
+import { ButtonDataItem } from './button-data-item';
+import { FormDataItem } from './form-data-item';
+import { FormDataItemType } from './form-data-item';
+
+export class IntegerDataItem extends FormDataItem {
+  placeholder: string;
+  display: string;
+  validationSettings: IntegerValidationSettings;
+
+  constructor(key: string,
+              label: string,
+              tooltip: string,
+              helpText: string,
+              required: boolean,
+              pathParts: string[],
+              type: FormDataItemType,
+              value: any,
+              isReadOnly: boolean,
+              isHidden: boolean,
+              public buttons: ButtonDataItem[],
+              schemaOptions: SchemaIntegerOptions) {
+    super(key, label, tooltip, helpText, required, pathParts, type, value, isReadOnly, isHidden);
+    this.placeholder = schemaOptions.placeholder;
+    this.display = schemaOptions.display;
+    this.validationSettings = {
+      length: { minLength: schemaOptions.minLength, maxLength: schemaOptions.maxLength },
+      range: { minimum: schemaOptions.minimum, exclusiveMinimum: schemaOptions.exclusiveMinimum,
+               maximum: schemaOptions.maximum, exclusiveMaximum: schemaOptions.exclusiveMaximum }
+    };
+  }
+}
+
+export interface IntegerLengthOptions {
+  minLength: number;
+  maxLength: number;
+}
+
+export interface IntegerRangeOptions {
+  minimum: number;
+  maximum: number;
+  exclusiveMaximum: number;
+  exclusiveMinimum: number;
+}
+
+export interface IntegerValidationSettings {
+  length: IntegerLengthOptions;
```

Comment:
Do we need length validation on numbers? I didn't see this in the PR description and I couldn't find this in the json schema docs for an integer.

Reply:
When I broke integer out of the string class, these keywords were not excluded for integer fields. I didn't want to break anything, but you're right, they are not part of the JSON schema definition for integers